src/app/shared/components/info/info.component.ts

```diff
@@ -1,9 +1,11 @@
-import { Component } from '@angular/core';
+import { Component, Input } from "@angular/core";
 
 @Component({
-  selector: 'app-info',
-  templateUrl: './info.component.html',
-  styleUrls: ['./info.component.scss']
+  selector: "app-info[title]",
```

Comment:
What is the reason that you added [title] in here?

Reply:
https://stackoverflow.com/questions/35528395/make-directive-input-required/54240378#54240378

I followed the advice here. Apparently this is the way to make an @Input required in Angular 15 and below. Not sure if I applied it correctly. The task description said that only the 'text' @Input is optional, I assumed that this means that 'title' is required.

Reply:
Hmmm... It is always good to learn something new. I was not aware of this required input notation. However, I doubt that this is a best practice - see, from v16 you can mark it with `Input({ required: true})` which is clear and straightforward. 
I would definitely check the existence of the input in `OnChanges` hook and assign a default value if not present.
Anyway, since task description is open to interpretation, it think this is fine. I mean if 'text' is optional, then it does not necessarily mean that 'title' is required.
But i found a little contradiction here. So, the selector marks the title required, but you made it optional with the ? notation a few lines below.